src/Umbraco.Web.UI.Client/src/packages/documents/documents/rollback/entity-action/rollback.action.ts

```diff
@@ -1,14 +1,21 @@
 import { UMB_ROLLBACK_MODAL } from '../constants.js';
 import { UmbEntityActionBase } from '@umbraco-cms/backoffice/entity-action';
-import { umbOpenModal } from '@umbraco-cms/backoffice/modal';
+import { UMB_MODAL_MANAGER_CONTEXT } from '@umbraco-cms/backoffice/modal';
 import { UMB_NOTIFICATION_CONTEXT } from '@umbraco-cms/backoffice/notification';
 import { UmbLocalizationController } from '@umbraco-cms/backoffice/localization-api';
 
 export class UmbRollbackDocumentEntityAction extends UmbEntityActionBase<never> {
 	#localize = new UmbLocalizationController(this);
 
 	override async execute() {
-		await umbOpenModal(this, UMB_ROLLBACK_MODAL, {});
+		const modalManagerContext = await this.getContext(UMB_MODAL_MANAGER_CONTEXT);
+		if (!modalManagerContext) return;
+
+		const modalContext = modalManagerContext.open(this, UMB_ROLLBACK_MODAL, {});
+
+		const data = await modalContext.onSubmit().catch(() => undefined);
+		if (!data) return;
+
```

Comment:
This needs to be v16-ified to use `umbOpenModal`.

Reply:
Yes, I agree, but it is not a merge problem, But I can make another PR for that :-)

Reply:
Looks like this was already done in `v16/dev` actually, so I'll re-apply.